### PR TITLE
BaSP TG-31: Timesheet components implementation

### PR DESCRIPTION
**Pull request by Baldizan**
Shared Table, Button, Form, Inputs and Modal implementation into timesheet screens.

### CODE REVIEW

**Review by Martinzancocchia — APPROVED** [review]

**Review by LucianaPrat — APPROVED** [review]

File: src/Components/Shared/Input/index.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ export const Input = (props) => {
         placeholder={placeholder}
         type={type}
         value={value}
+        required={required}
```
Comment (martumoser) [inline src/Components/Shared/Input/index.js:19]: This correction is okey but take in mind that you are modifying a shared component which is used on other parts of the code so be careful when you do this type of changes. Check with the team this change and maybe you have to create a hotfix pr to solve this and implement this change

**Review by martumoser — DISMISSED** [review]
Why you are adding an empty file on Table row shared component?

File: src/Components/TimeSheets/TimeSheetsForm/index.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    }));
+
+  return (
+    <section className={styles.container}>
+      <Form onSubmit={onSubmit} title={titleForm} noValidate={invalid} secondColumnIndex={3}>
+        <Select
+          onChange={onChange}
+          value={timeSheetInput.project}
+          name="project"
+          arrayToMap={project()}
```
Comment (martumoser) [inline src/Components/TimeSheets/TimeSheetsForm/index.js:137]: project() means that you are executing the function. You have to send project with execute it

File: src/Components/TimeSheets/TimeSheetsTable/index.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        data={list.map((row) => ({
+          ...row,
+          date: row.date.slice(0, 10),
+          project: row.project?._id,
+          task: row.task?._id,
+          employee: row.employee?._id,
+          projectName: row.project?.name ?? 'N/A',
+          taskDescription: row.task?.description ?? 'N/A',
+          employeeFormat: row.employee ? `${row.employee?.name} ${row.employee?.lastName}` : 'N/A'
+        }))}
```
Comment (martumoser) [inline src/Components/TimeSheets/TimeSheetsTable/index.js:67]: move this logic outside the return to have a clean code

**Review by martumoser — CHANGES_REQUESTED** [review]

**Review by NicolasGulam — APPROVED** [review]

**Pull request merged**